File: src/data/courses/dsa/string-hashing-problems/hashing-based-problems/longest-consecutive-sequence/code/cpp/longest_consecutive_sequence_sol1.cpp
class Solution {
public:
    // Sort: O(n log n) — Sort then scan consecutive runs
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        sort(nums.begin(), nums.end());
        int maxLen = 1, curLen = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] == nums[i-1]) continue;       // Skip duplicates
            if (nums[i] == nums[i-1] + 1) curLen++;   // Extend streak
            else curLen = 1;                            // Reset
            maxLen = max(maxLen, curLen);
        }
        return maxLen;
    }
};

File: src/data/courses/dsa/string-hashing-problems/hashing-based-problems/longest-consecutive-sequence/code/python/longest_consecutive_sequence_sol1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # Sort: O(n log n) — Sort then scan
        if not nums: return 0
        nums.sort()
        max_len = cur = 1
        for i in range(1, len(nums)):
            if nums[i] == nums[i-1]: continue
            if nums[i] == nums[i-1]+1: cur += 1
            else: cur = 1
            max_len = max(max_len, cur)
        return max_len